packages/components/src/public/QueryModel/QueryModel.ts

```diff
@@ -1093,4 +1138,50 @@ export class QueryModel {
     }
 }
 
-type QueryModelURLState = Pick<QueryModel, 'filterArray' | 'offset' | 'schemaQuery' | 'selectedReportId' | 'sorts'>;
+type QueryModelURLState = Pick<
+    QueryModel,
+    'filterArray' | 'maxRows' | 'offset' | 'schemaQuery' | 'selectedReportId' | 'sorts'
+>;
+type QueryModelSettings = Partial<Pick<QueryModel, 'filterArray' | 'maxRows' | 'sorts' | 'viewName'>>;
+const LOCAL_STORAGE_PREFIX = 'QUERY_MODEL_SETTINGS.';
+
+function localStorageKey(modelId: string): string {
+    return LOCAL_STORAGE_PREFIX + modelId;
+}
+
+export function getSettingsFromLocalStorage(id: string): QueryModelSettings {
+    const savedSettings = JSON.parse(localStorage.getItem(localStorageKey(id)));
+
+    if (savedSettings === null) return undefined;
+
+    const { maxRows, viewName } = savedSettings;
+    const filterArray = savedSettings.filterArray?.map(f =>
+        Filter.create(f.columnName, f.value, Filter.getFilterTypeForURLSuffix(f.type))
+    );
+    const sorts = savedSettings.sorts?.map(s => querySortFromString(s));
+
+    return {
+        filterArray: filterArray ?? [],
+        maxRows: maxRows ?? DEFAULT_MAX_ROWS,
+        sorts: sorts ?? [],
+        viewName,
```

Comment:
does viewName need a default? I assume not and undefined means "default view"?

Reply:
We only update the model's schemaQuery if viewName is defined, see withQueryModels line 179